docs/sources/flow/reference/components/discovery.docker.md

```diff
@@ -0,0 +1,167 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/discovery.docker
+title: discovery.docker
+---
+
+# discovery.docker
+
+`discovery.docker` discovers [Docker Engine][] containers and exposes them as targets.
+
+[Docker Engine]: https://docs.docker.com/engine/
+
+## Usage
+
+```river
+discovery.docker "LABEL" {
+  host = "DOCKER_ENGINE_HOST"
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`host` | `string` | Address of the Docker Daemon to connect to. | | yes
+`port` | `number` | Default port to collect metrics from when a port mapping isn't defined. | `80` | no
+`host_networking_host` | `string` | Host to use if the container is in host networking mode. | `"localhost"` | no
+`refresh_interval` | `duration` | Frequency to refresh list of containers. | `"1m"` | no
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`discovery.docker`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+filter | [filter][] | Filters discoverable resources. | no
+http_client_config | [http_client_config][] | HTTP client configuration for docker requests. | no
+http_client_config > basic_auth | [basic_auth][] | Configure basic_auth for authenticating to the endpoint. | no
+http_client_config > authorization | [authorization][] | Configure generic authorization to the endpoint. | no
+http_client_config > oauth2 | [oauth2][] | Configure OAuth2 for authenticating to the endpoint. | no
+http_client_config > oauth2 > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
+
+The `>` symbol indicates deeper levels of nesting. For example,
+`http_client_config > basic_auth` refers to a `basic_auth` block defined inside
+an `http_client_config` block.
+
+[filter]: #filter-block
+[http_client_config]: #http_client_config-block
+[basic_auth]: #basic_auth-block
+[authorization]: #authorization-block
+[oauth2]: #oauth2-block
+[tls_config]: #tls_config-block
+
+### filter block
+
+The `filter` block configures a filter to pass to the Docker Engine to limit
+the amount of containers returned. The `filter` block can be specified multiple
+times to provide more than one filter.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`name` | `string` | Filter name to use. | | yes
+`values` | `list(string)` | Values to pass to the filter. | | yes
+
+Refer to [List containers][List containers] from the Docker Engine API
+documentation for the list of supported filters and their meaning.
+
+[List containers]: https://docs.docker.com/engine/api/v1.41/#tag/Container/operation/ContainerList
```

Comment:
I've chosen to link out to this page because users can update to a new version of the Docker Engine API independent of Grafana Agent, so we risk being out of sync for anything we document here.